src/OpenTelemetry/Trace/ParentBasedSampler.cs

```diff
@@ -41,8 +41,7 @@ public ParentBasedSampler(Sampler delegateSampler)
         public override SamplingResult ShouldSample(in SamplingParameters samplingParameters)
         {
             var parentContext = samplingParameters.ParentContext;
-            if (/* TODO: TraceId is always provided due to AutoGenerateRootContextTraceId. That is being removed in RC1 and this can be put back.
-                 parentContext.TraceId == default ||*/ parentContext.SpanId == default)
+            if (parentContext.TraceId == default)
```

Comment:
I think when I put that comment in there I imagined it ending up:
```csharp
if (parentContext.TraceId == default || parentContext.SpanId == default)
```
But it might be overly cautious & unnecessary to also check SpanId?

Reply:
I think checking traceid alone is sufficient...